src/components/OverlayWrapper/OverlayWrapper.jsx

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import _ from 'lodash';
+import { lucidClassNames } from '../../util/style-helpers';
+import { createClass, getFirst, rejectTypes, omitProps } from '../../util/component-types';
+import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
+
+const cx = lucidClassNames.bind('&-OverlayWrapper');
+
+const {
+	bool,
+	node,
+	oneOf,
+	string,
+} = React.PropTypes;
+
+/**
+ *
+ * {"categories": ["utility"]}
+ *
+ * A wrapper with optional overlay to wrap content. `Overlay` is meant for overlaying an entire page, while this component is meant to wrap another component and cover its content.
+ *
+ */
+const OverlayWrapper = createClass({
```

Comment:
not sure about this name, not a big fan of `-Wrapper`, but there's already an `Overlay`, which is like page global. Maybe `OverlayAnchored` or something?